crates/red_knot_python_semantic/src/types/class.rs

```diff
@@ -792,13 +792,10 @@ impl<'db> ClassLiteral<'db> {
         }
 
         let new_function_symbol = self_ty
```

Comment:
similarly, this would be clearer if it were named `dunder_new_symbol` or similar